src/wasm-linker.cpp

```diff
@@ -210,6 +210,15 @@ void Linker::layout() {
     auto* func = out.wasm.getFunction(name);
     func->type = ensureFunctionType(getSig(func), &out.wasm)->name;
   }
+
+  // Export malloc and free whenever availble. JavsScript version of malloc has
```

Comment:
I'm not sure this is the best approach. As discussed in https://github.com/kripken/emscripten/pull/4469, emcc will export malloc and free, the only reason this doesn't work for wasm is that wasm doesn't have a prefix `_`. Solving that core issue would probably fix/avoid some other issues as well in the wasm-backend path.

But, if we prefer to get this in for now as a workaround and figure things out later, I'm ok with that.